File: backend/src/main/java/com/team01/backend/global/security/CustomAuthenticationEntryPoint.java
package com.team01.backend.global.security;

import com.team01.backend.global.response.ApiResponse;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.AuthenticationEntryPoint;
import org.springframework.stereotype.Component;
import tools.jackson.databind.ObjectMapper;

import java.io.IOException;

@Component
@RequiredArgsConstructor
public class CustomAuthenticationEntryPoint implements AuthenticationEntryPoint { // 인증 실패 관리

    // 필드 주입 -> 생성자 주입으로 수정
    private final ObjectMapper objectMapper;

    @Override
    public void commence(HttpServletRequest request,
                         HttpServletResponse response,
                         AuthenticationException ex) throws IOException {

        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
        response.setContentType("application/json;charset=UTF-8");

        ApiResponse<Void> apiResponse =
                ApiResponse.ofFailure("UNAUTHORIZED", "인증이 필요합니다.");

        response.getWriter().write(
                objectMapper.writeValueAsString(apiResponse)
        );
    }
}

File: backend/src/main/java/com/team01/backend/global/security/CustomAuthenticationEntryPoint.kt
package com.team01.backend.global.security

import com.team01.backend.global.response.ApiResponse
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.AuthenticationEntryPoint
import org.springframework.stereotype.Component
import com.fasterxml.jackson.databind.ObjectMapper

@Component
class CustomAuthenticationEntryPoint(
    private val objectMapper: ObjectMapper
) : AuthenticationEntryPoint {  // 인증 실패 관리

    override fun commence(
        request: HttpServletRequest,
        response: HttpServletResponse,
        ex: AuthenticationException
    ) {
        response.status = HttpServletResponse.SC_UNAUTHORIZED
        response.contentType = "application/json;charset=UTF-8"

        val apiResponse = ApiResponse.ofFailure<Void>("UNAUTHORIZED", "인증이 필요합니다.")
        response.writer.write(objectMapper.writeValueAsString(apiResponse))
    }

}